crates/core/src/models/mod.rs

```diff
@@ -234,3 +239,60 @@ impl ConvertibleStarknetBlock for BlockWithTxs {
         Ok(block.into())
     }
 }
+
+pub struct StarknetEvent(Event);
+
+impl StarknetEvent {
+    pub fn new(sn_event: Event) -> Self {
+        Self(sn_event)
+    }
+}
+
+#[async_trait]
+impl ConvertibleStarknetEvent for StarknetEvent {
+    async fn to_eth_log(&self, _client: &dyn KakarotClient) -> Result<Log, KakarotClientError> {
+        // let starknet_block_id = StarknetBlockId::Tag(BlockTag::Latest);
+        // let address = client.get_evm_address(&self.0.from_address, &starknet_block_id).await.unwrap();
+        // Convert the StarkNet address to a byte array
+        let sn_bytes = self.0.from_address.clone().to_bytes_be();
+        // Take the last 20 bytes of the StarkNet address to create the Ethereum address
+        let address = Address::from_slice(&sn_bytes[12..32]);
+        let topics: Vec<H256> = self.0.keys
+        // Splits the keys into chunks of 2 elements each.
+            .chunks(2)
+            .map(|chunk| {
+                // Converts the first element of the chunk to bytes and then to a BigUint.
+                // BigUint gives us arbitrary precision, to avoid prime field wrap arounds
+                let low = BigUint::from_bytes_be(&chunk[0].to_bytes_be());
+                // If there is a second element in the chunk, it's converted to bytes and then to a BigUint. 
+                // Otherwise, a BigUint is created from 0u128.
+                let high = chunk.get(1).map_or(BigUint::from(0u128), |h| BigUint::from_bytes_be(&h.to_bytes_be()));
```

Comment:
the original python logic is written as a try/catch, also 
https://github.com/kkrt-labs/kakarot/blob/2133aaf58d5c8ae493c579570e43c9e011774309/tests/utils/contracts.py#L119

so it seems like we need to fail when the high value doesn't exist or if the bytes len is less than 32